### PR TITLE
Fix some statement orderings in scalc

### PR DESCRIPTION
**Pull request by AltGr**
~~Note: queued behind #607~~
EDIT: rebased

### CODE REVIEW

File: compiler/scalc/from_lcalc.ml [hunk 1]
```diff
@@ -49,7 +49,40 @@ exception NotAnExpr of { needs_a_local_decl : bool }
 (** Contains the LocalDecl of the temporary variable that will be defined by the
     next block is it's here *)
 
-let rec translate_expr (ctxt : 'm ctxt) (expr : 'm L.expr) : A.block * A.expr =
+(** Blocks are constructed as reverse ordered lists. This module abstracts this
+    and avoids confusion in ordering of statements (also opening the opportunity
+    for more optimisations) *)
+module RevBlock : sig
```
Comment (denismerigoux) [inline compiler/scalc/from_lcalc.ml:55]: <3 <3 <3

**Comment by AltGr** [comment]
Also added in a small fix to Clerk for passing more `-I` parameters to the OCaml compiler ([1412e76](https://github.com/CatalaLang/catala/pull/608/commits/1412e76938972159219ab8b79fee17e6ee829d87)) and avoid a warning

**Review by denismerigoux — APPROVED** [review]
Thanks Louis for making my code saner :)

**Review by denismerigoux — APPROVED** [review]
Of for this commit too!

File: compiler/scalc/from_lcalc.ml
[the remark this answers is not in the document]
Comment (AltGr) [reply compiler/scalc/from_lcalc.ml:1]: It's nice to have this fixed now before it bites us on the C backend :)

**Pull request merged**